From c1bd34adbd203b0d3ab7b3c3ada43f017b450c22 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Tue, 5 Dec 2023 11:12:38 -0600
Subject: [PATCH] solus: Set default vconsole font

Set ter-v32b as the fallback font if no FONT= stanza is active in
/etc/vconsole.conf

Signed-off-by: Reilly Brogan <reilly@reillybrogan.com>
---
 src/vconsole/vconsole-setup.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/vconsole/vconsole-setup.c b/src/vconsole/vconsole-setup.c
index 28df5e6ac5..673fe32ebc 100644
--- a/src/vconsole/vconsole-setup.c
+++ b/src/vconsole/vconsole-setup.c
@@ -474,7 +474,12 @@ int main(int argc, char **argv) {
         (void) toggle_utf8_sysfs(utf8);
         (void) toggle_utf8_vc(vc, fd, utf8);
 
-        r = font_load_and_wait(vc, vc_font, vc_font_map, vc_font_unimap);
+        /* Solus default setting. Use ter-v32b as the default tty font */
+        char default_font[] = "ter-v32b";
+        if (vc_font)
+                r = font_load_and_wait(vc, vc_font, vc_font_map, vc_font_unimap);
+        else
+                r = font_load_and_wait(vc, default_font, vc_font_map, vc_font_unimap);
         keyboard_ok = keyboard_load_and_wait(vc, vc_keymap, vc_keymap_toggle, utf8) == 0;
 
         if (idx > 0) {
-- 
2.43.0

